account_partner_aged_statement_webkit/report/partner_aged_statement_report.py

```diff
@@ -0,0 +1,392 @@
+                        journal_ids = journal_obj.search(
+                            self.cr, self.uid,
+                            [('type', 'in', ['bank', 'cash'])],
+                            context=self.localcontext)
+
+                    journal_id = journal_ids[0]
+
+                    lines_in_currency = voucher_obj.recompute_voucher_lines(
+                        self.cr, self.uid, ids=False, partner_id=partner.id,
+                        journal_id=journal_id, price=0,
+                        currency_id=invoice.currency_id.id,
+                        ttype=self.ttype, date=date,
+                        context=self.localcontext)['value']
+
+                    other_currencies[invoice.currency_id.id] = {
+                        'lines_in_currency': {
+                            line['move_line_id']: line for line in
+                            lines_in_currency['line_cr_ids']
+                            + lines_in_currency['line_dr_ids']
+                        },
+                        'move_line_ids': [move_line.id],
+                    }
+
+        for currency_id in other_currencies:
+            for move_line_id in other_currencies[currency_id]['move_line_ids']:
+                line_dict[move_line_id] = other_currencies[currency_id][
+                    'lines_in_currency'][move_line_id]
+
+        for line in line_dict:
+            move_line = move_line_obj.browse(
+                self.cr, self.uid, line_dict[line]['move_line_id'],
+                context=self.localcontext)
+            line_dict[line]['ref'] = move_line.ref
+
+            currency = currency_obj.browse(
+                self.cr, self.uid, line_dict[line]['currency_id'],
+                context=self.localcontext)
+            line_dict[line]['currency_name'] = currency.name
+
+        # Adjust the amount signs depending on the report type
+        if self.ttype == 'receipt':
+            for line in line_dict:
+                if line_dict[line]['type'] == 'dr':
+                    line_dict[line]['amount_original'] *= -1
+                    line_dict[line]['amount_unreconciled'] *= -1
+
+        elif self.ttype == 'payment':
+            for line in line_dict:
+                if line_dict[line]['type'] == 'cr':
+                    line_dict[line]['amount_original'] *= -1
+                    line_dict[line]['amount_unreconciled'] *= -1
+
+        # Order the invoices by date and check for lines with no amount
+        invoice_list = [
+            inv for inv in
+            sorted(line_dict.values(), key=lambda x: x['date_original'])
+            if inv['amount_original'] or inv['amount_unreconciled']
+        ]
+
+        # Order from the most recent to the older invoice.
+        invoice_list.reverse()
+
+        self.partner_invoices_dict[partner.id] = invoice_list
+
+        return self.partner_invoices_dict[partner.id]
+
+    def _lines_get_30(self, partner, company):
+        today = datetime.now()
```

Comment:
Should this be set using the users timezone?  I guess it only affects people accessing servers in significantly different timezones to their own.